In [1]:
delta = 0.012
rho = 1.225
s = 0.05
A = 0.503
omega = 300
r = 0.4
U_tip = 120
k = 0.1
v_0 = 4.03
d_0 = 0.6

P_0 = (delta / 8) * rho * s * A * (omega**3) * (r ** 3)

mu_1 = P_0
mu_2 = P_0 *(3 / (U_tip ** 2))
mu_3 = P_0 * v_0
mu_4 = (1/2) * d_0 * rho * s * A

print("mu_1=", mu_1)
print("mu_2=", mu_2)
print("mu_3=", mu_3)
print("mu_4=", mu_4)

mu_1= 79.85628000000003
mu_2= 0.016636725000000005
mu_3= 321.82080840000015
mu_4= 0.009242624999999999


$$ E(u,d) = ​\mu_1\frac{d}{v}+\mu_2dv+\mu_3\frac{d}{v^2}+\mu_4dv^2 $$
$$= 79.8563\frac{d}{v}+0.0166dv+321.8208\frac{d}{v^2}+0.0092dv^2$$

In [5]:
def energy(v,d):
    
    mu_1= 79.85628000000003
    mu_2= 0.016636725000000005
    mu_3= 321.82080840000015
    mu_4= 0.009242624999999999
    
    return mu_1 * d/v + mu_2 * d*v + mu_3 * d/(v**2) + mu_4 * d * (v**2) 

import math
d=200*math.sqrt(2)
energy(20, d)

2496.695196781387

In [ ]:


for i in range(3):
    for j in range(3):
        
        position = (i, j)
        node = i + j * 3  # unique node id

        if i==0 and j==0:
            base = node  
            time_window=[0,0]
        elif i == 1: 
            time_window = [1000, 2000]
        else:
            time_window = [0, 2000]

        print(position, time_window)

(0, 0) [0, 0]
(0, 1) [0, 2000]
(0, 2) [0, 2000]
(1, 0) [1000, 2000]
(1, 1) [1000, 2000]
(1, 2) [1000, 2000]
(2, 0) [0, 2000]
(2, 1) [0, 2000]
(2, 2) [0, 2000]


In [3]:
tour = [3,6,1,5]
tour[0:-1]

[3, 6, 1]

In [2]:
import sys
sys.path.insert(0, str('/Users/kirtisoglu/Documents/GitHub/UAV-Routing/uav_routing'))